### ISSUE TITLE
Nvidia 470 is it making any difference in GPU offload in hybrid mode?

### ISSUE THREAD

**Issue by michaldybczak** [Askannz/optimus-manager#420]
Nvidia 470 drivers were released, and they are already available in Manjaro testing. All works fine on my end with hybrid mode.

Since there is a note that there are some Optimus improvements, I wonder, if this driver changes anything regarding GPU offload in hybrid mode?

**Comment by XRaTiX** [comment]
I think the improvements was on the Intel only mode side,because in Intel mode only I can now use the NVIDIA card,not hybrid mode requirement,and when I'm done the NVIDIA turn off by itself (1050 Ti).

There is still some problems,applications using the NVIDIA doesn't sync with the refresh rate of the monitor,and have a little worse performance,and also cannot cannot execute 32-bit OpenGL games with the NVIDIA card (it gives me an error),but this is a beta driver so i'm pretty sure this will be fixed.

**Comment by michaldybczak** [comment]
How did you access Nvidia in Intel mode? Or are you talking about Nouveau, which worked for me for a long time in Intel session?

**Comment by XRaTiX** [comment]
No,I'm talking about the official NVIDIA drivers,not Nouveau.

And I just upgraded and switch to Intel mode,I don't have to do anything,in optimus-manager just make sure you are not using any switching methods,this is are my settings (using optimus-manager-qt),in fact you don't need optimus-manager at all and will it work (but then you cannot change to hybrid/nvidia mode to be able to use the HDMI port,so you kind of need it).

![image](https://user-images.githubusercontent.com/40683158/125200670-a84fa980-e231-11eb-8981-1016db6ca8c2.png)


Here is nvidia-smi in Intel mode
![image](https://user-images.githubusercontent.com/40683158/125200554-4e4ee400-e231-11eb-8a12-e0fe485bb5df.png)

And when I execute vkcube with the NVIDIA card.

![image](https://user-images.githubusercontent.com/40683158/125200622-7dfdec00-e231-11eb-9952-fe5e9d6f3fec.png)

**Comment by geminis3** [comment]
> No,I'm talking about the official NVIDIA drivers,not Nouveau.
> 
> And I just upgraded and switch to Intel mode,I don't have to do anything,in optimus-manager just make sure you are not using any switching methods,this is are my settings (using optimus-manager-qt),in fact you don't need optimus-manager at all and will it work (but then you cannot change to hybrid/nvidia mode to be able to use the HDMI port,so you kind of need it).
> 
> ![image](https://user-images.githubusercontent.com/40683158/125200670-a84fa980-e231-11eb-8981-1016db6ca8c2.png)
> 
> Here is nvidia-smi in Intel mode ![image](https://user-images.githubusercontent.com/40683158/125200554-4e4ee400-e231-11eb-8a12-e0fe485bb5df.png)
> 
> And when I execute vkcube with the NVIDIA card.
> 
> ![image](https://user-images.githubusercontent.com/40683158/125200622-7dfdec00-e231-11eb-9952-fe5e9d6f3fec.png)
> 
> EDIT: Actually I remember that I had to do something but with TLP,I need to make sure to have SOUND_POWER_SAVE_ON_AC to 1 (is 0 by default),without sound power save,the NVIDIA card still remains on and cannot powered off in Intel mode,I can check if the NVIDIA is on by running the command
> 
> `watch -n1 -d -t cat /sys/bus/pci/devices/0000:01:00.0/power/runtime_status`
> 
> If is ON,it says active, otherwise it says suspended

Doesn't works for me, GTX1060 on Acer G3-571 (Arch Linux).

**Comment by XRaTiX** [comment]
> > No,I'm talking about the official NVIDIA drivers,not Nouveau.
> > And I just upgraded and switch to Intel mode,I don't have to do anything,in optimus-manager just make sure you are not using any switching methods,this is are my settings (using optimus-manager-qt),in fact you don't need optimus-manager at all and will it work (but then you cannot change to hybrid/nvidia mode to be able to use the HDMI port,so you kind of need it).
> > ![image](https://user-images.githubusercontent.com/40683158/125200670-a84fa980-e231-11eb-8981-1016db6ca8c2.png)
> > Here is nvidia-smi in Intel mode ![image](https://user-images.githubusercontent.com/40683158/125200554-4e4ee400-e231-11eb-8a12-e0fe485bb5df.png)
> > And when I execute vkcube with the NVIDIA card.
> > ![image](https://user-images.githubusercontent.com/40683158/125200622-7dfdec00-e231-11eb-9952-fe5e9d6f3fec.png)
> > EDIT: Actually I remember that I had to do something but with TLP,I need to make sure to have SOUND_POWER_SAVE_ON_AC to 1 (is 0 by default),without sound power save,the NVIDIA card still remains on and cannot powered off in Intel mode,I can check if the NVIDIA is on by running the command
> > `watch -n1 -d -t cat /sys/bus/pci/devices/0000:01:00.0/power/runtime_status`
> > If is ON,it says active, otherwise it says suspended
> 
> Doesn't works for me, GTX1060 on Acer G3-571 (Arch Linux).

Do you set to integrated mode in optimus manager?

This should work,someone on the Manjaro forums says it works exactly this way but with CUDA applications

https://forum.manjaro.org/t/configuring-hybrid-gpu-laptop-to-only-use-dgpu-when-requested/76265/20?u=xratix

Unless is Manjaro specific configuration i don't know

**Comment by geminis3** [comment]
> 
> Do you set to integrated mode in optimus manager?
> 
> This should work,someone on the Manjaro forums says it works exactly this way but with CUDA applications
> 
> https://forum.manjaro.org/t/configuring-hybrid-gpu-laptop-to-only-use-dgpu-when-requested/76265/20?u=xratix
> 
> Unless is Manjaro specific configuration i don't know

When I boot the OS the dGPU is suspended but as soon as it enables then it can't suspend again.

Well afterall seems more convenient to use the nouveau method when I don't need dGPU.

**Comment by XRaTiX** [comment]
> > Do you set to integrated mode in optimus manager?
> > This should work,someone on the Manjaro forums says it works exactly this way but with CUDA applications
> > https://forum.manjaro.org/t/configuring-hybrid-gpu-laptop-to-only-use-dgpu-when-requested/76265/20?u=xratix
> > Unless is Manjaro specific configuration i don't know
> 
> When I boot the OS the dGPU is suspended but as soon as it enables then it can't suspend again.
> 
> Well afterall seems more convenient to use the nouveau method when I don't need dGPU.

I had the same problem,its TLP that prevents the dGPU to turn off again,you can either disable TLP or check your TLP config file  `/etc/tlp.conf` check 

```
SOUND_POWER_SAVE_ON_AC="1"
SOUND_POWER_SAVE_ON_BAT=1
```

Both needs to have a value of 1

**Comment by geminis3** [comment]
> > > Do you set to integrated mode in optimus manager?
> > > This should work,someone on the Manjaro forums says it works exactly this way but with CUDA applications
> > > https://forum.manjaro.org/t/configuring-hybrid-gpu-laptop-to-only-use-dgpu-when-requested/76265/20?u=xratix
> > > Unless is Manjaro specific configuration i don't know
> > 
> > When I boot the OS the dGPU is suspended but as soon as it enables then it can't suspend again.
> > 
> > Well afterall seems more convenient to use the nouveau method when I don't need dGPU.
> 
> I had the same problem,its TLP that prevents the dGPU to turn off again,you can either disable TLP or check your TLP config file  `/etc/tlp.conf` check 
> 
> ```
> SOUND_POWER_SAVE_ON_AC="1"
> SOUND_POWER_SAVE_ON_BAT=1
> ```
> 
> Both needs to have a value of 1
> 
> Then restart TLP and it should be suspended the dGPU
> 
> `sudo tlp start`
> 

I already have that in my config as per your previous post.

**Comment by XRaTiX** [comment]
> > > > Do you set to integrated mode in optimus manager?
> > > > This should work,someone on the Manjaro forums says it works exactly this way but with CUDA applications
> > > > https://forum.manjaro.org/t/configuring-hybrid-gpu-laptop-to-only-use-dgpu-when-requested/76265/20?u=xratix
> > > > Unless is Manjaro specific configuration i don't know
> > > 
> > > 
> > > When I boot the OS the dGPU is suspended but as soon as it enables then it can't suspend again.
> > > Well afterall seems more convenient to use the nouveau method when I don't need dGPU.
> > 
> > 
> > I had the same problem,its TLP that prevents the dGPU to turn off again,you can either disable TLP or check your TLP config file  `/etc/tlp.conf` check
> > ```
> > SOUND_POWER_SAVE_ON_AC="1"
> > SOUND_POWER_SAVE_ON_BAT=1
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Both needs to have a value of 1
> > Then restart TLP and it should be suspended the dGPU
> > `sudo tlp start`
> 
> I already have that in my config as per your previous post.

Try to check if both of these are `auto` too,maybe thats necessary too.

```
RUNTIME_PM_ON_AC=auto
RUNTIME_PM_ON_BAT=auto
```

**Comment by saikat0511** [comment]
@XRaTiX I am getting the same hybrid mode like experience (minus the persistent 4Mib xorg process in nvidia-smi) with the default integrated mode settings on arch linux and no TLP installed.
![image](https://user-images.githubusercontent.com/57610828/140072468-40ffb591-7e65-477f-b99f-0ca0a8333b5c.png)

**Comment by XRaTiX** [comment]
> @XRaTiX I am getting the same hybrid mode like behavior (minus the persistent 4Mib xorg process in nvidia-smi) with the default integrated mode settings on arch linux and no TLP installed. Does it make this mode better than the real "hybrid" mode? thinking ![image](https://user-images.githubusercontent.com/57610828/140072468-40ffb591-7e65-477f-b99f-0ca0a8333b5c.png)

Its definitely much better,you don't have a process consuming your NVIDIA card for nothing (4 MB is not alot,but why have it in the first place) and the NVIDIA card is powered off when is not being used.

So now the NVIDIA card behave like Windows,power off when not needed and can invoke it when I want.

https://user-images.githubusercontent.com/40683158/140162443-893c5729-fab1-43a6-b4f6-936a0f0fbeda.mp4

The only downside is the HDMI port still not working when you are in integrated mode as I said,[somebody on Reddit](https://www.reddit.com/r/linux_gaming/comments/qks7vp/the_nvidia_495_driver_and_the_gbm_support_lets_me/) said that in Wayland the HDMI port is working,so in the future there is no gonna be integrated/hybrid/NVIDIA mode anymore,will just work just like in Windows.

**Comment by michaldybczak** [comment]
On GTX 970M HDMI port works without problems in hybrid mode (not attached to GPU thou).

**Comment by XRaTiX** [comment]
> Btw how did you get gpu suspension to work? arch wiki and the nvidia page mentioned it only works on turing and ampere GPUs so I just accepted that it wont work for my system (MX330, optimus laptop. icelake cpu so that's supported atleast). It just stays active all the time.
> 
> Going by your previous posts you have a 1050ti (pascal) so technically it shouldn't work for you.

Yes I have a 1050 Ti,but to be honest I don't even know why it works,just make TLP `RUNTIME_PM_ON_AC` enabled (on) but even sometimes that doesn't work (see the previous comments above,was helping the guy and somehow it doesn't work for him),I don't know if it something related with Manjaro that does the power management and in Arch is not present.

One thing for sure it the recent nvidia-utils package for 495,my NVIDIA card was `Active` all the time like you and I found out that the culprit is a file located `/usr/share/egl/egl_external_platform.d/15_nvidia_gbm.json`,if I delete that file and reboot,my NVIDIA card is able to poweroff again,I explain more details here

https://gitlab.manjaro.org/packages/extra/nvidia-utils/-/issues/2#note_24337

So try deleting that file to see if your card power off too.

> On GTX 970M HDMI port works without problems in hybrid mode (not attached to GPU thou).

Yeah the HDMI port works in hybrid,I was talking about in Integrated mode the HDMI port doesn't work unless the port is attached to your iGPU (And this is solved in Wayland).

**Comment by saikat0511** [comment]
> One thing for sure it the recent nvidia-utils package for 495,my NVIDIA card was `Active` all the time like you and I found out that the culprit is a file located `/usr/share/egl/egl_external_platform.d/15_nvidia_gbm.json`,if I delete that file and reboot,my NVIDIA card is able to poweroff again,I explain more details here
> 
> https://gitlab.manjaro.org/packages/extra/nvidia-utils/-/issues/2#note_24337
> 
> So try deleting that file to see if your card power off too.
> 
> >

Didn't work for me :(

**Comment by geminis3** [comment]
If somebody is interested I made a simple CLI tool (GUI soon) to effectively turn OFF and ON the Nvidia dGPU on laptops using Udev rules.

Nvidia ON:

![Captura de pantalla de 2021-11-04 18-55-16](https://user-images.githubusercontent.com/49411169/140436934-812fde5e-c828-4bc3-ace9-85ed208e4510.png)


Nvidia OFF:

![Captura de pantalla de 2021-11-04 18-56-54](https://user-images.githubusercontent.com/49411169/140436923-ed67dbad-1267-4925-8216-262219b9110a.png)

This tool supports both Nouveau and Nvidia propietary drivers, it should work on any Linux distro.

**Comment by I-dont-need-name** [comment]
> No,I'm talking about the official NVIDIA drivers,not Nouveau.
> 
> And I just upgraded and switch to Intel mode,I don't have to do anything,in optimus-manager just make sure you are not using any switching methods,this is are my settings (using optimus-manager-qt),in fact you don't need optimus-manager at all and will it work (but then you cannot change to hybrid/nvidia mode to be able to use the HDMI port,so you kind of need it).
> 
> ![image](https://user-images.githubusercontent.com/40683158/125200670-a84fa980-e231-11eb-8981-1016db6ca8c2.png)
> 
> Here is nvidia-smi in Intel mode ![image](https://user-images.githubusercontent.com/40683158/125200554-4e4ee400-e231-11eb-8a12-e0fe485bb5df.png)
> 
> And when I execute vkcube with the NVIDIA card.
> 
> ![image](https://user-images.githubusercontent.com/40683158/125200622-7dfdec00-e231-11eb-9952-fe5e9d6f3fec.png)
> 
> EDIT: Actually I remember that I had to do something but with TLP,I need to make sure to have SOUND_POWER_SAVE_ON_AC to 1 (is 0 by default),without sound power save,the NVIDIA card still remains on and cannot powered off in Intel mode,I can check if the NVIDIA is on by running the command
> 
> `watch -n1 -d -t cat /sys/bus/pci/devices/0000:01:00.0/power/runtime_status`
> 
> If is ON,it says active, otherwise it says suspended

But can you run other applications with nvidia GPU? And can you run nvidia settings? I have the same result but is seems like without xorg process nothing can be ran.

**Comment by XRaTiX** [comment]
> > No,I'm talking about the official NVIDIA drivers,not Nouveau.
> > And I just upgraded and switch to Intel mode,I don't have to do anything,in optimus-manager just make sure you are not using any switching methods,this is are my settings (using optimus-manager-qt),in fact you don't need optimus-manager at all and will it work (but then you cannot change to hybrid/nvidia mode to be able to use the HDMI port,so you kind of need it).
> > ![image](https://user-images.githubusercontent.com/40683158/125200670-a84fa980-e231-11eb-8981-1016db6ca8c2.png)
> > Here is nvidia-smi in Intel mode ![image](https://user-images.githubusercontent.com/40683158/125200554-4e4ee400-e231-11eb-8a12-e0fe485bb5df.png)
> > And when I execute vkcube with the NVIDIA card.
> > ![image](https://user-images.githubusercontent.com/40683158/125200622-7dfdec00-e231-11eb-9952-fe5e9d6f3fec.png)
> > EDIT: Actually I remember that I had to do something but with TLP,I need to make sure to have SOUND_POWER_SAVE_ON_AC to 1 (is 0 by default),without sound power save,the NVIDIA card still remains on and cannot powered off in Intel mode,I can check if the NVIDIA is on by running the command
> > `watch -n1 -d -t cat /sys/bus/pci/devices/0000:01:00.0/power/runtime_status`
> > If is ON,it says active, otherwise it says suspended
> 
> But can you run other applications with nvidia GPU? And can you run nvidia settings? I have the same result but is seems like without xorg process nothing can be ran.

Yes,I can use any application with the NVIDIA gpu,see the video I posted.

https://github.com/Askannz/optimus-manager/issues/420#issuecomment-959772366

NVIDIA settings doesn't work tho unless I switch to hybrid or nvidia mode for some reason,but to me it doesn't matter I never use that application.

**Comment by molitona** [comment]
> If somebody is interested I made a simple CLI tool (GUI soon) to effectively turn OFF and ON the Nvidia dGPU on laptops using Udev rules.
> 
> https://github.com/geminis3/EnvyControl
> 
> Nvidia ON:
> 
> ![Captura de pantalla de 2021-11-04 18-55-16](https://user-images.githubusercontent.com/49411169/140436934-812fde5e-c828-4bc3-ace9-85ed208e4510.png)
> 
> Nvidia OFF:
> 
> ![Captura de pantalla de 2021-11-04 18-56-54](https://user-images.githubusercontent.com/49411169/140436923-ed67dbad-1267-4925-8216-262219b9110a.png)
> 
> This tool supports both Nouveau and Nvidia propietary drivers, it should work on any Linux distro.

Hi @geminis3 does this tool is another tool for switching like optimus manager, and what's the diff between them ?

**Comment by geminis3** [comment]
> Hi @geminis3 does this tool is another tool for switching like optimus manager, and what's the diff between them ?

Hi, EnvyControl is a tool written from scratch in Python that allows optimus-manager like graphics switching. The `integrated` mode works by blacklisting all Nvidia related modules as well as removing the Nvidia card from the PCI bus allowing good battery savings.

Unlike optimus-manager, EnvyControl is not only compatible with Arch Linux (it's available on the AUR) but should work with any other Linux distro (Debian derivatives require the initramfs to be rebuilt and EnvyControl does it automatically on newer versions).

You can read the FAQs [here](https://github.com/geminis3/envycontrol/wiki/Frequently-Asked-Questions).

**Comment by molitona** [comment]
@geminis3 
--------------

Sometimes, when I launch firefox, I get that black rectangle border.
https://i.ibb.co/TMLQFnT/d.png, so I've to kill/relaunch it.


I use Firefox 95.0.2 from Arch Repo.

When I launch firefox from terminal, I get that when getting that black rectangle border.
```
ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
!!! [Parent][RunMessage] Error: Channel closing: too late to send/recv, messages will be lost
```

When firefox starts normally, I get that instead:

```
ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
ATTENTION: default value of option mesa_glthread overridden by environment.
[GFX1-]: Failed to create EGLContext!: 0x3001
ATTENTION: default value of option mesa_glthread overridden by environment.
!!! [Parent][RunMessage] Error: Channel closing: too late to send/recv, messages will be lost
```
I noticed the same thing for telegram app, but I can't get a screen because it disappears fastly.
I just used optimus-manager to switch gpus with default config lastly when that problem started.
It does happen only on Xorg, wayland no.
I think it's something related to GPU and display.

Do you have any idea ?

**Comment by geminis3** [comment]
> @geminis3 
> --------------
> 
> Sometimes, when I launch firefox, I get that black rectangle border.
> https://i.ibb.co/TMLQFnT/d.png, so I've to kill/relaunch it.
> 
> 
> I use Firefox 95.0.2 from Arch Repo.
> 
> When I launch firefox from terminal, I get that when getting that black rectangle border.
> ```
> ATTENTION: default value of option mesa_glthread overridden by environment.
> ATTENTION: default value of option mesa_glthread overridden by environment.
> ATTENTION: default value of option mesa_glthread overridden by environment.
> ATTENTION: default value of option mesa_glthread overridden by environment.
> !!! [Parent][RunMessage] Error: Channel closing: too late to send/recv, messages will be lost
> ```
> 
> When firefox starts normally, I get that instead:
> 
> ```
> ATTENTION: default value of option mesa_glthread overridden by environment.
> ATTENTION: default value of option mesa_glthread overridden by environment.
> ATTENTION: default value of option mesa_glthread overridden by environment.
> ATTENTION: default value of option mesa_glthread overridden by environment.
> [GFX1-]: Failed to create EGLContext!: 0x3001
> ATTENTION: default value of option mesa_glthread overridden by environment.
> !!! [Parent][RunMessage] Error: Channel closing: too late to send/recv, messages will be lost
> ```
> I noticed the same thing for telegram app, but I can't get a screen because it disappears fastly.
> 
> I started after using optimus-manager to switch gpus with default config.
> It does happen only on Xorg, wayland no (obviously cause wayland not supported by optimus manager)
> I think it's something related to GPU and display.
> 
> Do you have any idea ?
> I'll change optimus-manager by envycontrol and tell you results, but before that I've to know what causes that problem after using optimus-manager (maybe it will reoccur with envycontrol ? dunno what's the root cause ?)

I had the same bug on Gnome Wayland using EnvyControl integrated mode, sometimes Firefox works and sometimes it gets frozen with black rectangles all over the screen.

It doesn't happens on Xorg for me.

**Comment by molitona** [comment]
@geminis3 
Why Isn't a logout sufficient to switch mode like optimus manager ?

**Comment by geminis3** [comment]
> @geminis3 
> Why Isn't a logout sufficient to switch mode like optimus manager ?

I think optimus-manager loads and unloads Nvidia modules whilst the system is running (as well as restarting Xorg), EnvyControl just creates the necessary files to tell the system to load this or blacklist this on next boot.

I can achieve the same results but I don't find it necessary with today's SSD and - - NVMe drives that boot up in just a couple of seconds.

**Comment by molitona** [comment]
@geminis3 
Not all people have SSD, like mine.
BTW, is there any command to know if HDMI ports are wired to the dGPU or IGPU ?

**Comment by geminis3** [comment]
> @geminis3 
> Not all people have SSD, like mine.
> BTW, is there any command to know if HDMI ports are wired to the dGPU or IGPU ?

If on `integrated` mode HDMI ports don't work then it's wired to the dGPU, also you can check the service manual for your laptop model mine has a system block diagram where it explicitly shows that as well that it's a muxless (mshybrid) design.

**Comment by molitona** [comment]
I've Inspiron 5558, I haven't found anything related to HDMI on manuel, could you help ?

**Comment by molitona** [comment]
> > > No,I'm talking about the official NVIDIA drivers,not Nouveau.
> > > And I just upgraded and switch to Intel mode,I don't have to do anything,in optimus-manager just make sure you are not using any switching methods,this is are my settings (using optimus-manager-qt),in fact you don't need optimus-manager at all and will it work (but then you cannot change to hybrid/nvidia mode to be able to use the HDMI port,so you kind of need it).
> > > ![image](https://user-images.githubusercontent.com/40683158/125200670-a84fa980-e231-11eb-8981-1016db6ca8c2.png)
> > > Here is nvidia-smi in Intel mode ![image](https://user-images.githubusercontent.com/40683158/125200554-4e4ee400-e231-11eb-8a12-e0fe485bb5df.png)
> > > And when I execute vkcube with the NVIDIA card.
> > > ![image](https://user-images.githubusercontent.com/40683158/125200622-7dfdec00-e231-11eb-9952-fe5e9d6f3fec.png)
> > > EDIT: Actually I remember that I had to do something but with TLP,I need to make sure to have SOUND_POWER_SAVE_ON_AC to 1 (is 0 by default),without sound power save,the NVIDIA card still remains on and cannot powered off in Intel mode,I can check if the NVIDIA is on by running the command
> > > `watch -n1 -d -t cat /sys/bus/pci/devices/0000:01:00.0/power/runtime_status`
> > > If is ON,it says active, otherwise it says suspended
> > 
> > 
> > But can you run other applications with nvidia GPU? And can you run nvidia settings? I have the same result but is seems like without xorg process nothing can be ran.
> 
> Yes,I can use any application with the NVIDIA gpu,see the video I posted.
> 
> [#420 (comment)](https://github.com/Askannz/optimus-manager/issues/420#issuecomment-959772366)
> 
> NVIDIA settings doesn't work tho unless I switch to hybrid or nvidia mode for some reason,but to me it doesn't matter I never use that application.

@XRaTiX  I've the same "active" problem on Integrated+Hybrid modes. Deleting that file doesn't work. I use 470.94 driver. Any solution ?

**Comment by XRaTiX** [comment]
> @XRaTiX I've the same "active" problem on Integrated+Hybrid modes. Deleting that file doesn't work. I use 470.94 driver. Any solution ?

I noticed that are some files that prevent the NVIDIA card from powering off,check if you have this files (at least on Manjaro,maybe Arch and other Arch based system have this too)

```
/usr/share/egl/egl_external_platform.d/15_nvidia_gbm.json

/usr/lib/udev/rules.d/60-nvidia.rules
```

If you have it delete them,the first file is for wayland and doesn't affect xorg,the second file I don't know what is but I have delete it and it doesn't affect my system,keep a backup just in case.

After that reboot and now it should say suspended instead of active,also make sure to have TLP blacklisted the NVIDIA card via PCIe,I have it like this in TLP UI

![image](https://user-images.githubusercontent.com/40683158/147878661-1896ecd4-a2cb-47e6-96e3-e2e6f4df51ca.png)

 After that is just like in the video,I can invoke it when I want and it powered off when not in use.

**Comment by molitona** [comment]
Thanks @XRaTiX 
I followed config 2 here to disable nvidia card on intel mode.
https://github.com/Askannz/optimus-manager/wiki/A-guide--to-power-management-options#configuration-2--pci-power-control

It worked for me by loading nouveau, so no nvidia driver anymore on lsmod output.
But, it's hacky solution, because when I launch firefox and some other programs, watch -n1 -d -t cat /sys/bus/pci/devices/0000:08:00.0/power/runtime_status out goes from suspended --> active --> suspending --> suspended. They are polling the nvidia driver.

So, I removed nouveau option and used pci_power_control=yes, pci_remove=yes as they said there. Now lsmod for nvidia or nouveau doesn't return anything.
